TKeazirian.HTTPServer.Tests/Router/RouterTests.cs

```diff
@@ -28,33 +32,45 @@ public void RouterRoutesToExpectedHandler()
     [InlineData("/")]
     public void ResourceNotFoundHandlerCalledWhenPathIsNotConfigured(string path)
     {
-        HTTPServer.Request.Request testRequest = new HTTPServer.Request.Request("GET", path, "", "");
-        var testRoutesConfig = new RoutesConfig(new Dictionary<string, HTTPServer.Handler.Handler>
+        Request testRequest = new Request("GET", path, "", "");
+        var testRoutesConfig = new RoutesConfig(new Dictionary<string, Handler>
         {
             { "/test_path", new MockHandler() }
         });
         Router router = new Router(testRoutesConfig);
 
-        HTTPServer.Response.Response response = router.Route(testRequest);
+        Response response = router.Route(testRequest);
 
         Assert.Equal("HTTP/1.1 404 Not Found\r\n", response.ResponseStatusLine);
     }
 
     [Fact]
     public void IfGetMethodIsAllowedOnEndpointThenHeadMethodIsAllowed()
     {
-        HTTPServer.Request.Request testRequest = new HTTPServer.Request.Request("HEAD", "/test_path", "", "");
+        Request testRequest = new Request("HEAD", "/test_path", "", "");
         string testHeaders = "Content-Type: text/plain\r\nContent-Length: 9\r\n\r\n";
-        var testRoutesConfig = new RoutesConfig(new Dictionary<string, HTTPServer.Handler.Handler>
+        var testRoutesConfig = new RoutesConfig(new Dictionary<string, Handler>
         {
             { "/test_path", new MockHandler() }
         });
         Router router = new Router(testRoutesConfig);
 
-        HTTPServer.Response.Response response = router.Route(testRequest);
+        Response response = router.Route(testRequest);
 
         Assert.Equal("HTTP/1.1 200 OK\r\n", response.ResponseStatusLine);
         Assert.Equal(testHeaders, response.ResponseHeaders);
         Assert.Empty(response.GetBody());
     }
+
+    [Fact]
+    public void IfImproperFormattedRequest404IsCalled()
+    {
+        Request badTestRequest = HelperFunctions.ImproperFormattedRequest("", "GET", "/test_path");
+        Router router = new Router(new RoutesConfig());
```

Comment:
Now that's an interesting point.  Should there be a default? and them test that the default works?

Reply:
In other words, right now it appears the the `ResourceNotFound` is required in the `ResourceConfig` .. should it be, optional?

Reply:
[Here](https://github.com/t-keazirian/HTTPServer/blob/b371b691287be3627c48746e0cd3bde201808698/TKeazirian.HTTPServer/Server/Router.cs#L27) is the point where your `Handler` is tightly coupled to the `Router`, which is tied to the `RouterConfig`.

I think this is relevant to this PR, because adding the test is bringing this issue to light.

Reply:
They are coupled because the `Router` calls the `Handler`?  Is that what that means?

Reply:
I like how this played out.